src/txn-wal/src/txns.rs

```diff
@@ -415,11 +420,9 @@ where
                 }
             }
 
-            // Note: Ordering here matters, we want to generate the Tidy work _before_ removing the
-            // handle because the work will create a handle to the shard.
```

Comment:
Is this no longer true? I remember generating the tidy work after removing the handle would prevent finalizing shards because `txn-wal` would end up holding a write handle indefinitely